libraries/AP_SmartAudio/AP_SmartAudio.h

```diff
@@ -0,0 +1,205 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
+#include <AP_Param/AP_Param.h>
+#include <stdio.h>
+
+#ifndef SMARTAUDIO_ENABLED
+#define SMARTAUDIO_ENABLED !HAL_MINIMIZE_FEATURES && !APM_BUILD_TYPE(APM_BUILD_Replay)
+#endif
+
+#if SMARTAUDIO_ENABLED
+#define SMARTAUDIO_BUFFER_CAPACITY 5
+
+#include "smartaudio_protocol.h"
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <AP_HAL/utility/RingBuffer.h>
+#include <AP_RCTelemetry/AP_VideoTX.h>
+
+
+class AP_SmartAudio
+{
+
+public:
+
+    // request packet to be processed
+    struct packet{
+        smartaudioFrame_t frame;
+        uint8_t frame_size=0;
+        uint32_t sended_at=0;
+    } PACKED;
+
+    // configured external params
+    struct configured_smartaudio_params{
+        AP_Int8  enabled;
+        AP_Int8  protocol_version;
+        AP_Int8  setup_defaults;
+    } smartaudio_params;
+
+    AP_SmartAudio();
+
+    static AP_SmartAudio *get_singleton(void)
+    {
+        return singleton;
+    }
+
+    /* Do not allow copies */
+    AP_SmartAudio(const AP_SmartAudio &other) = delete;
+
+    AP_SmartAudio &operator=(const AP_SmartAudio&) = delete;
+
+    static const struct AP_Param::GroupInfo var_info[];
+
+    // init threads and lookup for io uart.
+    bool init();
+
+    // looping over request loop
+    void loop();
+
+    // updates the smartaudio state in sync whith AP_VideoTX
+    bool update(bool force);
+
+    // RingBuffer to store outgoing request.
+    ObjectBuffer<packet> requests_queue{SMARTAUDIO_BUFFER_CAPACITY};
```

Comment:
This will use a lot of flash since this type has not been used before. It would be nice if we could make this common with the runcam packet, but probably not for this PR

Reply:
Why the ringbuffer is a flash eater ? It's seems as a empowered array.